### PR TITLE
docs: custom runner usage

### PR DESCRIPTION
**Pull request by aarnphm**
adding one API for `bentoml.Model.load_model` to initialize the model object.

This would fix some of the current bug

### CODE REVIEW

File: docs/source/concepts/runner.rst [hunk 1]
```diff
@@ -189,8 +189,45 @@ and used in the same service. For example:
...
-.. TODO::
-    Document creating custom Runnable with models from BentoML model store
+
+Custom Runnable built with Model from BentoML's model store:
+
+.. code::
+
+    from __future__ import annotations
+
+    from typing import Any
```
Comment (sauyon) [inline docs/source/concepts/runner.rst:199]:
```suggestion
    from typing import Any
```

File: docs/source/concepts/runner.rst [hunk 2]
```diff
@@ -189,8 +189,45 @@ and used in the same service. For example:
...
+
+    bento_model = bentoml.sklearn.get("spam_detection:latest")
+
+
+    class SpamDetectionRunnable(bentoml.Runnable):
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            # load the model instance
+            self.classifier = bento_model.load_model()
```
Comment (sauyon) [inline docs/source/concepts/runner.rst:215]: Let's just use this API so we don't have multiple ways to load a model from the bento model?
```suggestion
            self.classifier = bentoml.sklearn.load_model(bento_model)
```

File: docs/source/concepts/runner.rst [hunk 3]
```diff
@@ -189,8 +189,45 @@ and used in the same service. For example:
...
+        def is_spam(self, input_data: ArrayLike) -> ArrayLike:
+            return self.classifier.predict(input_data)
+
+
+    spam_detection_runner = bentoml.Runner(SpamDetectionRunnable, models=[bento_model])
+
+
+    svc = bentoml.Service("spam_detector", runners=[spam_detection_runner])
+
+
```
Comment (sauyon) [inline docs/source/concepts/runner.rst:227]:
```suggestion

    spam_detection_runner = bentoml.Runner(SpamDetectionRunnable, models=[bento_model])
    svc = bentoml.Service("spam_detector", runners=[spam_detection_runner])

```

File: docs/source/concepts/runner.rst [hunk 4]
```diff
@@ -189,8 +189,45 @@ and used in the same service. For example:
...
+    import bentoml
+    from bentoml.io import JSON
+    from bentoml.io import NumpyNdarray
+    from numpy.typing import ArrayLike
+
+
+    bento_model = bentoml.sklearn.get("spam_detection:latest")
+
+
```
Comment (sauyon) [inline docs/source/concepts/runner.rst:209]: Not personally a huge fan of all the double spaces, especially in a short example.

```suggestion

    bento_model = bentoml.sklearn.get("spam_detection:latest")

```

File: docs/source/concepts/runner.rst [hunk 5]
```diff
@@ -189,8 +189,45 @@ and used in the same service. For example:
...
+    class SpamDetectionRunnable(bentoml.Runnable):
+        SUPPORTS_CPU_MULTI_THREADING = True
```
Comment (sauyon) [inline docs/source/concepts/runner.rst:211]: For future thought: should we have something like

```python
bentoml_models = (
    bentoml.sklearn.get("spam_detection:latest"),
)
```

that automatically passes the required models through to the runner?

File: bentoml/_internal/runner/runner.py [hunk 6]
```diff
@@ -108,8 +108,13 @@ def __init__(
...
         else:
             config = runners_config
-
-        models = [] if models is None else models
+        if models is None:
+            models = []
+        else:
+            if not all(isinstance(model, Model) for model in models):
+                raise ValueError(
+                    f"models must be a list of 'bentoml.Model' instances. Got {[type(model) for model in models]} instead."
```
Comment (aarnphm) [inline bentoml/_internal/runner/runner.py:116]:
```suggestion
                    f"models must be a list of 'bentoml.Model'. Got {set([type(model) for model in models if not isinstance(model, Model)])} instead."
```

File: docs/source/concepts/runner.rst [hunk 7]
```diff
@@ -189,8 +189,38 @@ and used in the same service. For example:
...
+Custom Runnable built with Model from BentoML's model store:
+
+.. code::
+
+    from typing import Any
+
+    import bentoml
+    from bentoml.io import JSON
+    from bentoml.io import NumpyNdarray
+    from numpy.typing import ArrayLike
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:202]:
```suggestion
    from numpy.typing import NDArray
```

File: docs/source/concepts/runner.rst [hunk 8]
```diff
@@ -189,8 +189,38 @@ and used in the same service. For example:
...
+
+    class SpamDetectionRunnable(bentoml.Runnable):
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            # load the model instance
+            self.classifier = bentoml.sklearn.load_model(bento_model)
+
+        @bentoml.Runnable.method(batchable=False)
+        def is_spam(self, input_data: ArrayLike) -> ArrayLike:
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:214]:
```suggestion
        def is_spam(self, input_data: NDArray[Any]) -> NDArray[Any]:
```

File: docs/source/concepts/runner.rst [hunk 9]
```diff
@@ -189,8 +189,38 @@ and used in the same service. For example:
...
+            return self.classifier.predict(input_data)
+
+
+    spam_detection_runner = bentoml.Runner(SpamDetectionRunnable, models=[bento_model])
+    svc = bentoml.Service("spam_detector", runners=[spam_detection_runner])
+
+    @svc.api(input=NumpyNdarray(), output=JSON())
+    def analysis(input_text: ArrayLike) -> dict[str, Any]:
```
Comment (aarnphm) [inline docs/source/concepts/runner.rst:222]:
```suggestion
    def analysis(input_text: NDArray[Any]) -> dict[str, Any]:
```

File: bentoml/_internal/models/model.py [hunk 10]
```diff
@@ -301,6 +305,11 @@ def to_runnable(self) -> t.Type[Runnable]:
             self._runnable = self.info.imported_module.get_runnable(self)
         return self._runnable
 
+    def load_model(self, *args: t.Any, **kwargs: t.Any) -> t.Any:
+        if self._model is None:
+            self._model = self.info.imported_module.load_model(self, *args, **kwargs)
+        return self._model
+
```
Comment (aarnphm) [inline bentoml/_internal/models/model.py:312]:
```suggestion
```

**Review by ssheng — APPROVED** [review]

**Pull request merged**